azure-pipelines.yml

```diff
@@ -47,16 +47,16 @@ jobs:
         PYTHON: '3.10'
         NUMPY: '1.25'
         CONDA_ENV: azure_ci
-        RUN_TYPEGUARD: yes
+        RUN_TYPEGUARD: no
```

Comment:
Note: this was discussed out of band. Typeguard broke in an unknown way that seemed unrelated to this PR. It is being switched off for now.